python/seldon_core/metrics.py

```diff
@@ -1,10 +1,115 @@
+from prometheus_client.core import (
+    HistogramMetricFamily,
+    GaugeMetricFamily,
+    CounterMetricFamily,
+    CollectorRegistry,
+)
+from prometheus_client import exposition
+from prometheus_client.utils import floatToGoString
+
+from multiprocessing import Manager
+
+import numpy as np
+
 from typing import List, Dict
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
 
 COUNTER = "COUNTER"
 GAUGE = "GAUGE"
 TIMER = "TIMER"
 
 
+def generate_metrics(metrics):
+    myregistry = CollectorRegistry()
+    myregistry.register(metrics)
+    return (
+        exposition.generate_latest(myregistry).decode("utf-8"),
+        exposition.CONTENT_TYPE_LATEST,
+    )
+
+
+BINS = [0] + list(np.logspace(-3, np.log10(30), 50)) + [np.inf]
```

Comment:
would be good to add a comment of what's happening here as there's quite a lot of np specific pieces

Reply:
Added comment.